src/config.rs

```diff
@@ -254,7 +263,15 @@ macro_rules! create_config {
         impl<'a> ConfigSetter<'a> {
             $(
             pub fn $i(&mut self, value: $ty) {
-                (self.0).$i.2 = value;
+                // we check if we are using nightly channel
```

Comment:
I don't think we need to check for nightly channel here - if a program is usinf Rustfmt as a library, then it is up to them to enforce the instability invariants.

Reply:
However, we should check this when we parse unstable_options from the toml file (there might be macro hygiene problems with this though).

Reply:
Haven't faced any problems with macro hygiene hopefully I've understood you correctly and is on the right track.